### DISCUSSION TITLE
Where does the RDBMS fit into the architecture?

### DISCUSSION THREAD

**Discussion by peg-leg** [Budibase/budibase#3617] [Help] [answered]
Looking at the architecture document: https://docs.budibase.com/references-1/budibase-architecture
I'm not sure where the relational DB fits in. What is stored in MS SQL or PostgreSQL or MariaDB as opposed to CouchDB when choosing one of those as the relational DB?

**Comment by mjashanks** [comment, marked as answer]
We don't package any relational DBs. Budibase can connect to these databases, but they must already exist somewhere else. CouchDB is packaged and stores all Budibase metadata, as well as "internal tables" aka Budibase DB.

In terms of architecture, the App Service (normal CRUD endpoints) or Worker Service (automations) will connect to these other DBs directly, once configured.

**Reply by peg-leg** [reply]
Excellent, understood.  So an external RDBMS can be used to store the individual app's data, but it's not bundled with Budibase.
